core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/AbstractFromConfigFactory.java

```diff
@@ -52,7 +59,7 @@ protected EntityDescriptor<Solution_> deduceEntityDescriptor(SolutionDescriptor<
         return entityDescriptor;
     }
 
-    protected EntityDescriptor<Solution_> deduceEntityDescriptor(SolutionDescriptor<Solution_> solutionDescriptor) {
+    protected EntityDescriptor<Solution_> getTheOnlyEntityDescriptor(SolutionDescriptor<Solution_> solutionDescriptor) {
```

Comment:
Is this called often, perhaps on the hot path? Because it is expensive. Collection size, creation of a new iterator... If it is on the hot path, may I suggest caching the result?

Reply:
No, `getTheOnlyEntityDescriptor()` is called only once or twice when the CH phase is being built. Flame Graph surface negligible, confirming it's not on the hot path.